src/vtk_calculator.py

```diff
@@ -18,34 +18,39 @@
 
 Scalar calculation and writing to given file
 
-./vtk_calculator.py inputmesh.vtk exp(cos(x)+sin(y)) -t e^(cos(x)+sin(y)) -o outputmesh.vtk
+vtk_calculator.py -m inputmesh.vtk -f "exp(cos(x)+sin(y))" --data "e^(cos(x)+sin(y))" -o outputmesh.vtk
 
-Vector field and appends to input mesh
+Vector field calculation and appends to input mesh
 
-./vtk_calculator.py inputmesh.vtk x*iHat+cos(y)*jHat-sin(z)*kHat -t MyVectorField
+vtk_calculator.py -m "inputmesh.vtk" -f "x*iHat+cos(y)*jHat-sin(z)*kHat" -d "MyVectorField"
 
 There is also a diff mode which provides statistic between input data and function calculated
 (Note that it only works for scalar data)
 
-./vtk_calculator.py -m inputmesh.vtu -f x+y -d mydata --diff --stats
+For example to calculate difference between given function "x+y"
+and existing data "sin(x)" and override the result to "sin(x)" and to save statistics into a file
+following command is used.
 
-Calculates difference between given function and mydata data save over rides into variable data and saves statistics
+vtk_calculator.py -m inputmesh.vtu -f "x+y" -d "sin(x)" --diff --stats
 
-./vtk_calculator.py -m inputmesh.vtu -f x+y -d diffence -diffd mydata --diff
-
-Calculates difference between given function and mydata data save into diffence data
+If you don't want to override "sin(x)" and prefer to save the newly generated
+data into another variable "difference" following command can be used.
 
+vtk_calculator.py -m inputmesh.vtu -f "x+y" -d "difference" --diffdata "sin(x)" --diff --stats
 """
 
 
 def parse_args():
     parser = argparse.ArgumentParser(description=__doc__)
-    parser.add_argument("--mesh", "-m", dest="in_meshname", required=True,
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("--mesh", "-m", dest="in_meshname",
                         help="The mesh (VTK Unstructured Grid) used as input")
-    parser.add_argument("--function", "-f", dest="function", required=True,
+    parser.add_argument("--function", "-f", dest="function", default="eggholder3d",
                         help="""The function to evalutate on the mesh.
             Syntax is the same as used in the calculator object, coordinates are given as e.g.  'cos(x)+y'.
-            Alternatively, you can use predefined function visit preCICE documentation.""")
+            Default is Egg-Holder function in 3D.
+            Alternatively, you can use predefined function """)
+    group.add_argument("--functions",action="store_true",help="Prints list of predefined functions.")
```

Comment:
We could make the action explicit: `--list-functions`.
Opinions on this?

Reply:
`--listfunctions` is a little bit long and arg parser does not allow `-` but I can change it to `--listfunctions` since it is not a frequently used option.